src/components/Metamask.js

```diff
@@ -35,4 +38,36 @@ const Metamask = ({}) => (
   </MetamaskWrapper>
 );
 
-export default Metamask;
+class LoadingMetamask extends Component {
+  render() {
+    if (this.props.web3.status === 'failed') {
+      if (this.props.errorComp) {
+        return <LoadingContainer {...this.props} />;
+      }
+      return <Metamask />;
+    }
+
+    if (
+      this.props.web3.status === 'initialized' &&
+      Object.keys(this.props.accounts).length === 0
+    ) {
+      return <Metamask />;
+    }
+
+    return <LoadingContainer {...this.props} />;
+  }
+}
+
+LoadingMetamask.contextTypes = {
+  drizzle: PropTypes.object,
+};
+
+const mapStateToProps = state => {
+  return {
+    accounts: state.accounts,
+    drizzleStatus: state.drizzleStatus,
```

Comment:
what for do we need the drizzleStatus here?

Reply:
We can take it off it's being used inside LoadingContainer